src/client/signatures.rs

```diff
@@ -86,3 +75,184 @@ impl From<CanonicalJSONError> for SignatureError {
         err.into()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{Collection, Verification};
+    use env_logger;
+    use httpmock::Method::GET;
+    use httpmock::{Mock, MockServer};
+    use serde_json::json;
+
+    fn verify_signature(
+        mock_server: &MockServer,
+        collection: Collection,
+        certificate: &str,
+        should_fail: bool,
+    ) {
+        let mut get_pem_certificate = Mock::new()
+            .expect_method(GET)
+            .expect_path(
+                "/chains/remote-settings.content-signature.mozilla.org-2020-09-04-17-16-15.chain",
+            )
+            .return_status(200)
+            .return_header("Content-Type", "application/json")
+            .return_body(certificate)
+            .create_on(&mock_server);
+
+        let mut verifiers: Vec<&dyn Verification> = Vec::new();
+
+        #[cfg(feature = "ring_verifier")]
+        verifiers.push(&super::ring_verifier::RingVerifier {});
```

Comment:
This looks a bit curious. You are creating a reference to a temporary here, and then push it to a vector. In general, a temporary only lives until the end of the statement. The only reason this works in this case is because the temporary implicitly beocmes a static constant. In case you'd ever add a verifier that needs an actual constructor call this pattern will fail.

The usual approach here would be to use owned trait objects, i.e. `Vec<Box<dyn Verification>>`. That way, the vector properly owns all the instances, and lifetimes can't become an issue.

The code is still fine – it works for the current use case, and you can still fix it if it ever becomes necessary, which is probably never. :)